langchain/src/tools/sql.ts

```diff
@@ -103,17 +110,22 @@ If there are any of the above mistakes, rewrite the query. If there are no mista
 
   llmChain: LLMChain;
 
-  constructor(llmChain?: LLMChain) {
+  constructor(llmChainOrOptions?: LLMChain | QueryCheckerToolArgs) {
     super();
-    if (llmChain) {
-      this.llmChain = llmChain;
+    if (typeof llmChainOrOptions?._chainType === "function") {
```

Comment:
CC @nfcampos saw this elsewhere to determine whether an object was a chain, but does this look ok with you?